tools/wake-format/emitter.cpp

```diff
@@ -1106,22 +1131,30 @@ wcl::doc Emitter::walk_binary(ctx_t ctx, CSTElement node) {
     parts = collect_right_binary(op_token, node);
   }
 
-  std::vector<wcl::optional<wcl::doc>> choices = {
+  if (!ctx.nested_binop &&
+      (op_token.id() == TOKEN_OP_DOLLAR ||
+       (op_token.id() == TOKEN_OP_OR && op_token.fragment().segment().str() == "|"))) {
```

Comment:
I think you should also check for a leading pipe instead of the whole string being a pipe. This gives us room to add operators in the future like `|>` and stuff.

Reply:
Confirming now, but I think the reason I had to do this was because `||` is also TOKEN_OP_OR which we don't want to hit this case

Reply:
Yep, confirmed. This matches `||` which looks not good

```
    def post = match _
        Pair (Pass output) Unit =
            if mode < 0
            || mode > 0x1ff then
                Fail (makeError "write {path}: Invalid mode ({strOctal mode})")
            else ....... cut ......
```

Reply:
ok yeah then just be conservative for now